### ISSUE TITLE
Resize that varies from batch to batch

### ISSUE THREAD

**Issue by carbocation** [kornia/kornia#2506]
## 🚀 Feature
I would like to vary the image size within an `AugmentationSequential` pipeline between batches. Of course all images in a batch must be resized to the same size, but from batch to batch there is no such dependency. For example, I would like to allow the X to vary between (180, 200) and the Y to vary between (190, 210). 

## Motivation

This will allow me to train with a variety of image sizes, to ensure that there is no accidental image size dependency in my training pipeline.

## Pitch

I'd like to be able to use something like Resize or RandomResizeCrop, but instead of fixing the image size at class instantiation, I'd like to fix the range, and have the actual size be determined during the forward pass.

## Alternatives

Alternatively, this can be done in item transforms prior to passing them in as a batch to Kornia.

## Additional context

I tried implementing this, but I did not see an example of an API where this type of random number generation is expected to happen in the forward pass (all of the examples seem to fix this in the init function), so I wasn't successful after a couple of quick tries. (In particular, since I need to take advantage of both the input and the mask augmentation for segmentation, simpler hacks aren't adequate.)

**Comment by shijianjian** [comment]
Hi @carbocation, can you give some code to demonstrate what kinds of function signature you are looking for?

I think it is better to define the input and output first.

**Comment by carbocation** [comment]
Thanks @shijianjian . In my mind, this class would be similar in many respects to `kornia.augmentation.RandomResizedCrop`.

If the current code for `RandomResizedCrop` looks like this:

```py
class kornia.augmentation.RandomResizedCrop(
size, 
scale=(0.08, 1.0), 
ratio=(3.0 / 4.0, 4.0 / 3.0), 
resample=Resample.BILINEAR.name, 
same_on_batch=False, 
align_corners=True, 
p=1.0, 
keepdim=False, 
cropping_mode='slice')
```

An example signature would swap out size for size_range and remove same_on_batch (because it would need to resize all images to the same size for any given batch - the size variation must be between batches). E.g.:

```py
class kornia.augmentation.RandomRandomResizedCrop(
size_range, 
scale=(0.08, 1.0), 
ratio=(3.0 / 4.0, 4.0 / 3.0), 
resample=Resample.BILINEAR.name, 
align_corners=True, 
p=1.0, 
keepdim=False, 
cropping_mode='slice')
```

Cloning from the current `RandomResizedCrop` help docs, I would like to change this:

```py
>>> rng = torch.manual_seed(0)
>>> inputs = torch.tensor([[[0., 1., 2.],
...                         [3., 4., 5.],
...                         [6., 7., 8.]]])
>>> aug = RandomResizedCrop(size=(3, 3), scale=(3., 3.), ratio=(2., 2.), p=1., cropping_mode="resample")
>>> out = aug(inputs)
>>> out
tensor([[[[1.0000, 1.5000, 2.0000],
          [4.0000, 4.5000, 5.0000],
          [7.0000, 7.5000, 8.0000]]]])
>>> aug.inverse(out, padding_mode="border")
tensor([[[[1., 1., 2.],
          [4., 4., 5.],
          [7., 7., 8.]]]])
```

To this:

```py
>>> rng = torch.manual_seed(0)
>>> inputs = torch.tensor([[[0., 1., 2.],
...                         [3., 4., 5.],
...                         [6., 7., 8.]]])
>>> aug = RandomRandomResizedCrop(size_range=((1,3), (2,3)), scale=(3., 3.), ratio=(2., 2.), p=1., cropping_mode="resample")
>>> out = aug(inputs)
>>> out
tensor([[[[1.0000, 1.5000, 2.0000],
          [4.0000, 4.5000, 5.0000],
          [7.0000, 7.5000, 8.0000]]]])
>>> aug.inverse(out, padding_mode="border")
tensor([[[[1., 1., 2.],
          [4., 4., 5.],
          [7., 7., 8.]]]])
```

I.e., changing `size=(3,3)` to `size_range=((1,3), (2,3))`. Then the output shape might be `(B, C, 1, 3)` for one batch and `(B, C, 3, 2)` for another, etc.

My primary motivation for starting with `RandomResizedCrop` is that I know it handles `['image']` and `['mask']` keys correctly. There may be other considerations that make a different class the better analogy, but this is how I first envisioned it.

**Comment by shijianjian** [comment]
I see. It looks like very intuitive to do and it is backward compatible.

No need to change any argument names, and only to support ``[tuple, tuple]`` input for the ``size`` argument, then ``RandomResizedCrop(size=((1,3), (2,3)))`` shall be fine.

So that, ``RandomResizedCrop(size=(3, (2,3)))`` and ``RandomResizedCrop(size=((2,3), 3))`` will be supported in the same time.

You may file up a PR if you got an implementation already.

**Comment by carbocation** [comment]
I took a look but wasn't able to figure out how to safely move [ResizedCropGenerator](https://github.com/kornia/kornia/blob/3c747ccc0b79d726e8d7c98b3c38e2a6c40b358b/kornia/augmentation/_2d/geometric/resized_crop.py#L76) out of the `__init__`.

**Comment by edgarriba** [comment]
If anyone is interested in this, feel free to re-open the ticket. Thanks!
